dockstore-webservice/src/main/java/io/dockstore/webservice/core/LimitedSourceFileBuilder.java

```diff
@@ -93,25 +94,34 @@ public SourceFile build() {
 
         private static void setContentWithLimits(SourceFile file, String content, String path) {
             String limitedContent = content;
-            if (content != null) {
+            SourceFile.FormEnum limitedForm = SourceFile.FormEnum.COMPLETE;
+            // Limit certain types of content.
+            if (content == null) {
+                limitedContent = "Dockstore could not retrieve this file";
+                limitedForm = SourceFile.FormEnum.ERROR;
+            } else {
                 byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
+                if (Bytes.indexOf(bytes, Byte.decode("0x00")) != -1) {
+                    // Postgres cannot store strings that contain "NUL" characters.
+                    // Thus, Dockstore cannot currently store binary files.
+                    // https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER
+                    // https://www.ascii-code.com/character/%E2%90%80
+                    limitedContent = "Dockstore does not store binary files";
+                    limitedForm = SourceFile.FormEnum.ERROR;
```

Comment:
Not exactly an error in this case since it is working as intended. 
Maybe "incomplete" or "placeholder"?

Reply:
> Not exactly an error in this case since it is working as intended. Maybe "incomplete" or "placeholder"?

I agree that this is not an error, it's the design, and I think it should be distinguishable from an actual error. The conversation is moving and some of this has been addressed the comments and in your PR description, so this is probably repetitive, but...

-  I also think it's worth distinguishing the cases where we don't have have the data because it's binary vs. the case where the content is too large (and your `TRUNCATED` proposal addresses that).